msal_extensions/token_cache.py

```diff
@@ -0,0 +1,193 @@
+"""Generic functions and types for working with a TokenCache that is not platform specific."""
+import os
+import sys
+import time
+import errno
+import msal
+from .cache_lock import CrossPlatLock
+
+if sys.platform.startswith('win'):
+    from .windows import WindowsDataProtectionAgent
+elif sys.platform.startswith('darwin'):
+    from .osx import Keychain
+
+
+def get_protected_token_cache(
+        cache_location=None,
+        lock_location=None,
+        enforce_encryption=False, **kwargs):
```

Comment:
From a usability standpoint, wouldn't it be somewhat misleading that such constructor's name is `get_protected_token_cache(...)`, yet one of the optional parameter is `enforce_encryption=False`? It would cause confusion for dev who reads our api signature and even its docs, but not reading its implementation (which means the majority of devs), "am I supposed to always enable `enforce_encryption` or what?".

Would better make the `encryption_thing=True` the default behavior. And then of course we would probably want to NOT blow out by `RuntimeError`. A noticeable warning would be fine. Please see my another comment for a more detailed proposal.

Reply:
Fixed in a68ba24616da7d9d810a5bc33b2d3d0ff3817244